search_that_hash/__main__.py

```diff
@@ -72,20 +78,149 @@ def main(**kwargs):
+                    continue
+                printing.Prettifier.error_print("No types found for this hash.", chash)
+            continue
+
+        hash_processes.append(cracking.Searcher.main(searcher, chash, types))
+
+        #### OUTPUTTING
+
+        chash = list(hash_processes[-1].keys())[0]
+        result = hash_processes[-1][chash]
+
+        if result == None and not config["greppable"]:
+            if config["api"]:
+                results.append({chash: "Could not crack hash"})
+                continue
+            printing.Prettifier.error_print("Could not crack hash.", chash)
+            continue
+
+        if not config["greppable"]:
+            if config["api"]:
+                results.append({chash: result})
+                continue
+            printing.Prettifier.one_print(chash, result)
+
+    if config["greppable"]:
+
+        #### ADDING STH RESULTS
+
+        for hash in hash_processes:
+            base_results = list(hash.values())[0]
+            try:
+                base = sth_results[list(hash.keys())[0]]
+
+                base_results[0].update({"STH_API": base["Plaintext"]})
+                base_results.append(
+                    {"Type": base["Type"], "Verified": base["Verified"]}
+                )
+                """ 
+
+				Firstly this takes the hash from the loop and gets is values
+
+				{hash:[{COMPLETED},{FAILED}]}
+
+				The [0] index is for the main list
+
+				The next [0] index is for the successful dicts
+
+				Then it updates it with STH which:
+
+				Gets all hashes in a list and indexs the hash which its looking for
+
+				Then it takes that plaintext and updates the successfull dict with
+
+				{'STH_API':'Result'}
+				"""
+
+            except:
+                base_results[1].update({"STH_API": "Failed"})
+                base_results.append({"Type": "Unkown", "Verified": "N/A"})
+
+                #### CHECKING IF HASH SHOULD BE PUSHED TO STH
+
+                types = []
+
+                for type_ in config["hashes"][list(hash.keys())[0]]:
+                    types.append(type_["name"])
+                    if len(types) == 4:
+                        break
+
+                if (
+                    "STH_API" not in list(base_results[0].keys())
+                    and base_results[0] != {}
+                ):
+                    sth.push(
+                        config,
+                        list(hash.keys())[0],
+                        list(base_results[0].values())[0],
+                        types,
+                    )
+                    # This pushes the config, hash, plaintext and types
+
+            # This is here in case STH does not crack it, in that case it does the exact same but refers to the fail list index (0 and not 1)
+
+        if not config["api"]:
+            logging.info("Printing greppable results")
+            printing.Prettifier.greppable_print(hash_processes)
+
+        return hash_processes
+
+    return results
+
+
```

Comment:
Is the function really this long?

Functions should:
* Do one and only one thing
Any more and it becomes mess :) Try to break it up into much smaller functions and put them into a module? :D